### ISSUE TITLE
Failed to apply mutation - std::_Nested_exception<schema_version_loading_failed> during restart_then_repair_node

### ISSUE THREAD

**Issue by fgelcer** [scylladb/scylladb#4648]
the job running is longevity 3h for 3.1, and for some reason, after the restart it failed to execute the repair

```
Jul 03 10:30:52 ip-172-30-0-15.ec2.internal scylla.bin[6381]:  [shard 0] storage_proxy - Failed to apply mutation from 172.30.0.176#0: std::_Nested_exception<schema_version_loading_failed> (Failed to load schema version 0a389c3d-22f9-306f-9af0-9f3ec85586dd): std::_Nested_exception<no_such_keyspace> (Can't find a keyspace keyspace1): std::out_of_range (_Map_base::at)
```

running with ami-084c44c9c845351bb (us-east-1)
[log from the node that had the failure](https://github.com/scylladb/scylla/files/3355560/database.log). if other node's logs are needed, they can be found on [here](https://cloudius-jenkins-test.s3.amazonaws.com/8dc6cdf6-d2af-43cf-9314-339f488a7b84/job-2019-07-03T09.24-b27349e.zip)

**Comment by roydahan** [comment]
Worth mentioning, it happened on a node that was trying to replace a dead node (itself in this case).
And eventually the schema version was synced and the node started successfully.

@fgelcer I don't see where the repair failed.
I see: 
`nodetool -h localhost repair" finished with status 0`

**Comment by tgrabiec** [comment]
Looks like a bootstrapping node receives writes from a coordinator before it completes bootstrap, before it syncs the schema. That's why keyspace definition is not found (that should be fixed as part of  https://github.com/scylladb/scylla/issues/3760).

The fact that coordinators contact a bootstrapping node is a bug, we should not allow this.

**Comment by slivne** [comment]
@tgrabiec so its clear this is not a regression - right ?

**Comment by tgrabiec** [comment]
@slivne I don't think it's a regression.

**Comment by bentsi** [comment]
Happened to me in `3.1.0.rc2-0.20190713.92bf92817` durin 2 nodetool cleanup ran in parallel on another node that was bootstrapping after deleting all of its data
```
Jul 16 12:33:06 ip-172-30-0-212.ec2.internal scylla.bin[2142]:  [shard 0] rpc - client 172.30.0.144:58506: server connection dropped: The TLS connection was non-properly terminated.
Jul 16 12:33:07 ip-172-30-0-212.ec2.internal scylla.bin[2142]:  [shard 0] gossip - InetAddress 172.30.0.226 is now UP, status = NORMAL
Jul 16 12:33:07 ip-172-30-0-212.ec2.internal scylla.bin[2142]:  [shard 0] gossip - InetAddress 172.30.0.45 is now UP, status = NORMAL
Jul 16 12:33:07 ip-172-30-0-212.ec2.internal scylla.bin[2142]:  [shard 2] compaction - Compacting [/var/lib/scylla/data/system/peers-37f71aca7dc2383ba70672528af04d4f/system-peers-ka-16-Data.db:level=0, /var/lib/scylla/data/system/peers-37f71aca7dc2383ba70672528af04d4f/system-peers-ka-30-Data.db:level=0, ]
Jul 16 12:33:07 ip-172-30-0-212.ec2.internal scylla.bin[2142]:  [shard 13] compaction - Compacting [/var/lib/scylla/data/system/peers-37f71aca7dc2383ba70672528af04d4f/system-peers-ka-27-Data.db:level=0, /var/lib/scylla/data/system/peers-37f71aca7dc2383ba70672528af04d4f/system-peers-ka-41-Data.db:level=0, ]
Jul 16 12:33:07 ip-172-30-0-212.ec2.internal scylla.bin[2142]:  [shard 11] compaction - Compacting [/var/lib/scylla/data/system/peers-37f71aca7dc2383ba70672528af04d4f/system-peers-ka-25-Data.db:level=0, /var/lib/scylla/data/system/peers-37f71aca7dc2383ba70672528af04d4f/system-peers-ka-39-Data.db:level=0, ]
Jul 16 12:33:07 ip-172-30-0-212.ec2.internal scylla.bin[2142]:  [shard 13] compaction - Compacted 2 sstables to [/var/lib/scylla/data/system/peers-37f71aca7dc2383ba70672528af04d4f/system-peers-ka-55-Data.db:level=0, ]. 15887 bytes to 11392 (~71% of original) in 8ms = 1.36MB/s. ~256 total partitions merged to 1.
Jul 16 12:33:07 ip-172-30-0-212.ec2.internal scylla.bin[2142]:  [shard 2] compaction - Compacted 2 sstables to [/var/lib/scylla/data/system/peers-37f71aca7dc2383ba70672528af04d4f/system-peers-ka-44-Data.db:level=0, ]. 15910 bytes to 11408 (~71% of original) in 9ms = 1.21MB/s. ~256 total partitions merged to 1.
Jul 16 12:33:07 ip-172-30-0-212.ec2.internal scylla.bin[2142]:  [shard 0] gossip - InetAddress 172.30.0.250 is now UP, status = NORMAL
Jul 16 12:33:07 ip-172-30-0-212.ec2.internal scylla.bin[2142]:  [shard 11] compaction - Compacted 2 sstables to [/var/lib/scylla/data/system/peers-37f71aca7dc2383ba70672528af04d4f/system-peers-ka-53-Data.db:level=0, ]. 15894 bytes to 11420 (~71% of original) in 9ms = 1.21MB/s. ~256 total partitions merged to 1.
Jul 16 12:33:07 ip-172-30-0-212.ec2.internal scylla.bin[2142]:  [shard 0] gossip - InetAddress 172.30.0.144 is now UP, status = NORMAL
Jul 16 12:33:07 ip-172-30-0-212.ec2.internal scylla.bin[2142]:  [shard 0] storage_proxy - Failed to apply mutation from 172.30.0.226#0: std::_Nested_exception<schema_version_loading_failed> (Failed to load schema version c1cf31b6-3972-3e61-bf7f-b5e89a421304): std::_Nested_exception<no_such_keyspace> (Can't find a keyspace keyspace_snappy): std::out_of_range (_Map_base::at)
Jul 16 12:33:16 ip-172-30-0-212.ec2.internal scylla.bin[2142]:  [shard 0] gossip - Gossip not settled after 1 polls.
Jul 16 12:33:19 ip-172-30-0-212.ec2.internal scylla.bin[2142]:  [shard 0] gossip - Gossip settled after 1 extra polls; proceeding
Jul 16 12:33:19 ip-172-30-0-212.ec2.internal scylla.bin[2142]:  [shard 0] gossip - Feature CORRECT_COUNTER_ORDER is enabled
Jul 16 12:33:19 ip-172-30-0-212.ec2.internal scylla.bin[2142]:  [shard 0] gossip - Feature CORRECT_NON_COMPOUND_RANGE_TOMBSTONES is enabled
Jul 16 12:33:19 ip-172-30-0-212.ec2.internal scylla.bin[2142]:  [shard 0] gossip - Feature CORRECT_STATIC_COMPACT_IN_MC is enabled
Jul 16 12:33:19 ip-172-30-0-212.ec2.internal scylla.bin[2142]:  [shard 0] gossip - Feature COUNTERS is enabled
Jul 16 12:33:19 ip-172-30-0-212.ec2.internal scylla.bin[2142]:  [shard 0] gossip - Feature DIGEST_INSENSITIVE_TO_EXPIRY is enabled
```

**Comment by slivne** [comment]
@tgrabiec - I thought https://github.com/scylladb/scylla/issues/4196 fixed the issue for regullar tables and bootstrap - e.g. a node will not join and not receive writes before it pulls the schema. What am I missing ?

**Comment by tgrabiec** [comment]
@slivne When adding a node which was not part of the cluster, that's enough, because existing nodes won't try to send anything to it before it tells them to. But here the bootstrapping node replaces an existing node which has the same IP, so existing nodes already know about it and maybe sending requests to it as soon as they manage to connect to it again.

If we want to support this scenario, then handling this correctly needs extra coordination.

**Comment by aleksbykov** [comment]
Got reproduction on latest scylla master:
**Scylla version 666.development-0.20200507.791acc7f3**
one node was decommissioned, new node was added to cluster. Log from new node:
[database.log](https://github.com/scylladb/scylla/files/4598182/database.log)

```
May 08 07:54:22 gemini-repo-with-nemesis-3h-normal--db-node-23389ed7-4 scylla[5457]:  [shard 1] compaction - Compacted 2 sstables to []. 10954 bytes to 0 (~0% of original) in 0ms = 0.00MB/s. ~256 total partitions merged to 0.
May 08 07:54:22 gemini-repo-with-nemesis-3h-normal--db-node-23389ed7-4 scylla[5457]:  [shard 0] init - starting messaging service
May 08 07:54:22 gemini-repo-with-nemesis-3h-normal--db-node-23389ed7-4 scylla[5457]:  [shard 0] repair - Setting max_repair_memory=719323136, max_repair_memory_per_range=33554432, max_repair_ranges_in_parallel=21
May 08 07:54:22 gemini-repo-with-nemesis-3h-normal--db-node-23389ed7-4 scylla[5457]:  [shard 0] init - starting storage service
May 08 07:54:22 gemini-repo-with-nemesis-3h-normal--db-node-23389ed7-4 systemd[1]: Started Scylla Server.
May 08 07:54:22 gemini-repo-with-nemesis-3h-normal--db-node-23389ed7-4 systemd[1]: Started Scylla JMX.
May 08 07:54:22 gemini-repo-with-nemesis-3h-normal--db-node-23389ed7-4 scylla[5457]:  [shard 1] compaction - Compacted 2 sstables to [/var/lib/scylla/data/system/truncated-38c19fd0fb863310a4b70d0cc66628aa/mc-7-big-Data.db:level=0, ]. 147465 bytes to 74121 (~50% of original) in 21ms = 3.37MB/s. ~256 total partitions merged to 6.
May 08 07:54:22 gemini-repo-with-nemesis-3h-normal--db-node-23389ed7-4 scylla[5457]:  [shard 0] storage_service - Loading persisted ring state
May 08 07:54:22 gemini-repo-with-nemesis-3h-normal--db-node-23389ed7-4 scylla[5457]:  [shard 0] storage_service - loaded_peer_features: peer_features size=0
May 08 07:54:22 gemini-repo-with-nemesis-3h-normal--db-node-23389ed7-4 scylla[5457]:  [shard 0] storage_service - Checking remote features with gossip
May 08 07:54:22 gemini-repo-with-nemesis-3h-normal--db-node-23389ed7-4 scylla[5457]:  [shard 0] messaging_service - Starting Messaging Service on port 7000
May 08 07:54:22 gemini-repo-with-nemesis-3h-normal--db-node-23389ed7-4 scylla[5457]:  [shard 0] compaction - Compacted 5 sstables to [/var/lib/scylla/data/system/local-7ad54392bcdd35a684174e047860b377/mc-38-big-Data.db:level=0, ]. 36780 bytes to 12800 (~34% of original) in 35ms = 0.35MB/s. ~640 total partitions merged to 1.
May 08 07:54:22 gemini-repo-with-nemesis-3h-normal--db-node-23389ed7-4 scylla[5457]:  [shard 0] compaction - Compacted 3 sstables to [/var/lib/scylla/data/system/truncated-38c19fd0fb863310a4b70d0cc66628aa/mc-12-big-Data.db:level=0, ]. 87207 bytes to 76909 (~88% of original) in 36ms = 2.04MB/s. ~384 total partitions merged to 2.
May 08 07:54:22 gemini-repo-with-nemesis-3h-normal--db-node-23389ed7-4 scylla-jmx[5473]: Picked up JAVA_TOOL_OPTIONS:
May 08 07:54:22 gemini-repo-with-nemesis-3h-normal--db-node-23389ed7-4 scylla-jmx[5473]: Using config file: /etc/scylla/scylla.yaml
May 08 07:54:22 gemini-repo-with-nemesis-3h-normal--db-node-23389ed7-4 scylla-jmx[5473]: Connecting to http://127.0.0.1:10000
May 08 07:54:22 gemini-repo-with-nemesis-3h-normal--db-node-23389ed7-4 scylla-jmx[5473]: Starting the JMX server
May 08 07:54:23 gemini-repo-with-nemesis-3h-normal--db-node-23389ed7-4 scylla[5457]:  [shard 0] gossip - Feature check passed. Local node 172.30.0.151 features = {CDC, COMPUTED_COLUMNS, CORRECT_COUNTER_ORDER, CORRECT_NON_COMPOUND_RANGE_TOMBSTONES, CORRECT_STATIC_COMPACT_IN_MC, COUNTERS, DIGEST_INSENSITIVE_TO_EXPIRY, DIGEST_MULTIPARTITION_READ, HINTED_HANDOFF_SEPARATE_CONNECTION, INDEXES, LARGE_PARTITIONS, LA_SSTABLE_FORMAT, LWT, MATERIALIZED_VIEWS, MC_SSTABLE_FORMAT, NONFROZEN_UDTS, PER_TABLE_PARTITIONERS, RANGE_TOMBSTONES, ROLES, ROW_LEVEL_REPAIR, SCHEMA_TABLES_V3, STREAM_WITH_RPC_STREAM, TRUNCATION_TABLE, UNBOUNDED_RANGE_TOMBSTONES, VIEW_VIRTUAL_COLUMNS, WRITE_FAILURE_REPLY, XXHASH}, Remote common_features = {CDC, COMPUTED_COLUMNS, CORRECT_COUNTER_ORDER, CORRECT_NON_COMPOUND_RANGE_TOMBSTONES, CORRECT_STATIC_COMPACT_IN_MC, COUNTERS, DIGEST_INSENSITIVE_TO_EXPIRY, DIGEST_MULTIPARTITION_READ, HINTED_HANDOFF_SEPARATE_CONNECTION, INDEXES, LARGE_PARTITIONS, LA_SSTABLE_FORMAT, LWT, MATERIALIZED_VIEWS, MC_SSTABLE_FORMAT, NONFROZEN_UDTS, PER_TABLE_PARTITIONERS, RANGE_TOMBSTONES, ROLES, ROW_LEVEL_REPAIR, SCHEMA_TABLES_V3, STREAM_WITH_RPC_STREAM, TRUNCATION_TABLE, UNBOUNDED_RANGE_TOMBSTONES, VIEW_VIRTUAL_COLUMNS, WRITE_FAILURE_REPLY, XXHASH}
May 08 07:54:23 gemini-repo-with-nemesis-3h-normal--db-node-23389ed7-4 scylla[5457]:  [shard 0] storage_service - Restarting a node in NORMAL status
May 08 07:54:23 gemini-repo-with-nemesis-3h-normal--db-node-23389ed7-4 scylla[5457]:  [shard 0] cdc - Restarting node in NORMAL status with CDC enabled, but no streams timestamp was proposed by this node according to its local tables. Are we upgrading from a non-CDC supported version?
May 08 07:54:23 gemini-repo-with-nemesis-3h-normal--db-node-23389ed7-4 scylla[5457]:  [shard 0] database - Schema version changed to 4a3ed0fa-af45-3f1c-8953-a7d7c9b27364
May 08 07:54:23 gemini-repo-with-nemesis-3h-normal--db-node-23389ed7-4 scylla[5457]:  [shard 0] storage_service - Starting up server gossip
May 08 07:54:23 gemini-repo-with-nemesis-3h-normal--db-node-23389ed7-4 scylla[5457]:  [shard 0] compaction - Compacting [/var/lib/scylla/data/system/local-7ad54392bcdd35a684174e047860b377/mc-40-big-Data.db:level=0, /var/lib/scylla/data/system/local-7ad54392bcdd35a684174e047860b377/mc-38-big-Data.db:level=0, ]
May 08 07:54:23 gemini-repo-with-nemesis-3h-normal--db-node-23389ed7-4 scylla[5457]:  [shard 0] compaction - Compacted 2 sstables to [/var/lib/scylla/data/system/local-7ad54392bcdd35a684174e047860b377/mc-42-big-Data.db:level=0, ]. 19404 bytes to 12349 (~63% of original) in 16ms = 0.74MB/s. ~256 total partitions merged to 1.
May 08 07:54:23 gemini-repo-with-nemesis-3h-normal--db-node-23389ed7-4 scylla-jmx[5473]: JMX is enabled to receive remote connections on port: 7199
May 08 07:54:24 gemini-repo-with-nemesis-3h-normal--db-node-23389ed7-4 scylla[5457]:  [shard 0] storage_proxy - Failed to apply mutation from 172.30.0.24#0: std::_Nested_exception<schema_version_loading_failed> (Failed to load schema version 8b649321-0865-3b06-931d-abb7318b33e8): std::_Nested_exception<no_such_keyspace> (Can't find a keyspace ks1): std::out_of_range (_Map_base::at)
May 08 07:54:24 gemini-repo-with-nemesis-3h-normal--db-node-23389ed7-4 scylla[5457]:  [shard 0] migration_manager - Requesting schema pull from 172.30.0.24:0
May 08 07:54:24 gemini-repo-with-nemesis-3h-normal--db-node-23389ed7-4 scylla[5457]:  [shard 0] migration_manager - Pulling schema from 172.30.0.24:0
May 08 07:54:24 gemini-repo-with-nemesis-3h-normal--db-node-23389ed7-4 scylla[5457]:  [shard 0] migration_manager - Requesting schema pull from 172.30.0.24:0
May 08 07:54:24 gemini-repo-with-nemesis-3h-normal--db-node-23389ed7-4 scylla[5457]:  [shard 0] migration_manager - Requesting schema pull from 172.30.0.24:0
May 08 07:54:24 gemini-repo-with-nemesis-3h-normal--db-node-23389ed7-4 scylla[5457]:  [shard 0] storage_proxy - Failed to apply mutation from 172.30.0.24#0: std::_Nested_exception<schema_version_loading_failed> (Failed to load schema version 8b649321-0865-3b06-931d-abb7318b33e8): std::_Nested_exception<no_such_keyspace> (Can't find a keyspace ks1): std::out_of_range (_Map_base::at)
May 08 07:54:24 gemini-repo-with-nemesis-3h-normal--db-node-23389ed7-4 scylla[5457]:  [shard 0] storage_proxy - Failed to apply mutation from 172.30.0.24#0: std::_Nested_exception<schema_version_loading_failed> (Failed to load schema version 8b649321-0865-3b06-931d-abb7318b33e8): std::_Nested_exception<no_such_keyspace> (Can't find a keyspace ks1): std::out_of_range (_Map_base::at)
May 08 07:54:24 gemini-repo-with-nemesis-3h-normal--db-node-23389ed7-4 scylla[5457]:  [shard 0] storage_proxy - Failed to apply mutation from 172.30.0.24#0: std::_Nested_exception<schema_version_loading_failed> (Failed to load schema version 8b649321-0865-3b06-931d-abb7318b33e8): std::_Nested_exception<no_such_keyspace> (Can't find a keyspace ks1): std::out_of_range (_Map_base::at)
May 08 07:54:24 gemini-repo-with-nemesis-3h-normal--db-node-23389ed7-4 scylla[5457]:  [shard 0] storage_proxy - Failed to apply mutation from 172.30.0.24#0: std::_Nested_exception<schema_version_loading_failed> (Failed to load schema version 8b649321-0865-3b06-931d-abb7318b33e8): std::_Nested_exception<no_such_keyspace> (Can't find a keyspace ks1): std::out_of_range (_Map_base::at)
May 08 07:54:24 gemini-repo-with-nemesis-3h-normal--db-node-23389ed7-4 scylla[5457]:  [shard 1] storage_proxy - Failed to apply mutation from 172.30.0.24#1: std::_Nested_exception<schema_version_loading_failed> (Failed to load schema version 8b649321-0865-3b06-931d-abb7318b33e8): std::_Nested_exception<no_such_keyspace> (Can't find a keyspace ks1): std::out_of_range (_Map_base::at)
May 08 07:54:24 gemini-repo-with-nemesis-3h-normal--db-node-23389ed7-4 scylla[5457]:  [shard 1] storage_proxy - Failed to apply mutation from 172.30.0.24#1: std::_Nested_exception<schema_version_loading_failed> (Failed to load schema version 8b649321-0865-3b06-931d-abb7318b33e8): std::_Nested_exception<no_such_keyspace> (Can't find a keyspace ks1): std::out_of_range (_Map_base::at)
May 08 07:54:24 gemini-repo-with-nemesis-3h-normal--db-node-23389ed7-4 scylla[5457]:  [shard 0] storage_proxy - Failed to apply mutation from 172.30.0.24#0: std::_Nested_exception<schema_version_loading_failed> (Failed to load schema version 8b649321-0865-3b06-931d-abb7318b33e8): std::_Nested_exception<no_such_keyspace> (Can't find a keyspace ks1): std::out_of_range (_Map_base::at)
May 08 07:54:24 gemini-repo-with-nemesis-3h-normal--db-node-23389ed7-4 scylla[5457]:  [shard 1] storage_proxy - Failed to apply mutation from 172.30.0.24#1: std::_Nested_exception<schema_version_loading_failed> (Failed to load schema version 8b649321-0865-3b06-931d-abb7318b33e8): std::_Nested_exception<no_such_keyspace> (Can't find a keyspace ks1): std::out_of_range (_Map_base::at)
May 08 07:54:24 gemini-repo-with-nemesis-3h-normal--db-node-23389ed7-4 scylla[5457]:  [shard 0] storage_proxy - Failed to apply mutation from 172.30.0.24#0: std::_Nested_exception<schema_version_loading_failed> (Failed to load schema version 8b649321-0865-3b06-931d-abb7318b33e8): std::_Nested_exception<no_such_keyspace> (Can't find a keyspace ks1): std::out_of_range (_Map_base::at)
```

**Comment by tgrabiec** [comment]
The fact that coordinators can contact replacing node early can lead to violation of consistency. The replacing node may not yet have all the writes its supposed to own.

This could theoretically violate linearizability of LWT.

**Comment by gleb-cloudius** [comment]
So why was it contacted? Was it declared as UP to early? Also the errors seams to be about writes. Writing into a bootstrapping node is normal.

**Comment by tgrabiec** [comment]
@aleksbykov In your case, the scenario is a bit different. 172.30.0.151 was first bootstrapped as a seed, then restarted. It looks like when it was down, its data directory was cleared, is that correct? Then after it booted, it was rebuilt (there is ` repair - rebuild_with_repair:` in the logs). 

In this case, it's expected that coordinator will contact this node, and you're only exposed to the schema sync problem, exposed by the fact that the node had its schema tables wiped out: https://github.com/scylladb/scylla/issues/3760

**Comment by juliayakovlev** [comment]
Found similar issue on ```2020.1.rc1-0.20200621.0659b8e6599```
longevity-lwt-24h test
Restart node with resharding nemesis.

Node 4 was restarted with resharding:
```
< t:2020-06-29 18:50:22,079 f:cluster.py      l:1651 c:sdcm.cluster_aws     p:DEBUG > Node longevity-lwt-24h-2020-1-db-node-2a10d859-4 [54.171.147.154 | 10.0.45.191] (seed: False): Change murmur3_partitioner_ignore_msb_bits to 15
```

After node was started and ran resharding, schema_version_loading_failed warning from node 6 that tried apply mutation from node 4:  
```< t:2020-06-29 18:51:13,442 f:cluster.py      l:1409 c:sdcm.cluster         p:DEBUG > 2020-06-29T18:51:02+00:00  longevity-lwt-24h-2020-1-db-node-2a10d859-4 !INFO    | scylla: [shard 0] init - Scylla version 2020.1.rc1-0.20200621.0659b8e6599 initialization completed.
< t:2020-06-29 18:54:34,669 f:cluster.py      l:1409 c:sdcm.cluster         p:DEBUG > 2020-06-29T18:54:20+00:00  longevity-lwt-24h-2020-1-db-node-2a10d859-6 !WARNING | scylla: [shard 6] storage_proxy - Failed to apply mutation from 10.0.45.191#6: std::_Nested_exception<schema_version_loading_failed> (Failed to load schema version 4eba1ade-92b4-30d3-ae74-82a4c449294c): std::runtime_error (Schema version 4eba1ade-92b4-30d3-ae74-82a4c449294c not found)
```

Test is still running and cluster is alive.
Jenkins job: ```https://jenkins.scylladb.com/view/enterprise-2020.1/job/enterprise-2020.1/job/longevity/job/longevity-lwt-24h/```
test_id: 2a10d859-6327-4aa4-9657-a292953b0aa4

**Comment by juliayakovlev** [comment]
Issue reproduced in branch 4.4

*Installation details*
Scylla version (or git commit hash): `4.4.rc2-0.20210224.49cd0b87f with build-id d1eabc8075f832eda38fe79f6d11072cc412b1f0`
Cluster size: 6 nodes (i3.4xlarge)
OS (RHEL/CentOS/Ubuntu/AWS AMI): `ami-021e046a074aed56d` (aws: eu-north-1)


Test: `longevity-cdc-4h-test`
Test name: `longevity_test.LongevityTest.test_custom_time`
Test config file(s):
- [longevity-cdc-100gb-4h.yaml](https://github.com/scylladb/scylla-cluster-tests/blob/ca8a02961c8a045df4e2ea49d1cc2fde06d11620/test-cases/longevity/longevity-cdc-100gb-4h.yaml)


Happened during restart_then_repair_node after restart of Scylla

```
< t:2021-02-25 16:06:24,423 f:remote_base.py  l:520  c:RemoteCmdRunner      p:DEBUG > Running command "sudo systemctl start scylla-server.service"...
< t:2021-02-25 16:06:28,101 f:base.py         l:140  c:RemoteCmdRunner      p:DEBUG > Command "sudo systemctl start scylla-server.service" finished with status 0
< t:2021-02-25 16:06:43,451 f:cluster.py      l:1395 c:sdcm.cluster         p:DEBUG > 2021-02-25T16:06:28+00:00  longevity-cdc-100gb-4h-4-4-db-node-6261f2f6-5 !WARNING | scylla: [shard 0] storage_proxy - Failed to apply mutation from 10.0.0.246#0: std::_Nested_exception<schema_version_loading_failed> (Failed to load schema version 7178e426-a1aa-32ab-98a1-fe0051b89949): std::_Nested_exception<no_such_keyspace> (Can't find a keyspace cdc_test): std::out_of_range (absl::container_internal::raw_hash_map<>::at)
```

Restore Monitor Stack command: `$ hydra investigate show-monitor 6261f2f6-1050-47ad-93c5-02015b6b0721`
Show all stored logs command: `$ hydra investigate show-logs 6261f2f6-1050-47ad-93c5-02015b6b0721`

Test id: `6261f2f6-1050-47ad-93c5-02015b6b0721`


Logs:
grafana - [https://cloudius-jenkins-test.s3.amazonaws.com/6261f2f6-1050-47ad-93c5-02015b6b0721/20210225_172323/grafana-screenshot-longevity-cdc-4h-test-scylla-per-server-metrics-nemesis-20210225_172712-longevity-cdc-100gb-4h-4-4-monitor-node-6261f2f6-1.png](https://cloudius-jenkins-test.s3.amazonaws.com/6261f2f6-1050-47ad-93c5-02015b6b0721/20210225_172323/grafana-screenshot-longevity-cdc-4h-test-scylla-per-server-metrics-nemesis-20210225_172712-longevity-cdc-100gb-4h-4-4-monitor-node-6261f2f6-1.png)
grafana - [https://cloudius-jenkins-test.s3.amazonaws.com/6261f2f6-1050-47ad-93c5-02015b6b0721/20210225_172323/grafana-screenshot-overview-20210225_172323-longevity-cdc-100gb-4h-4-4-monitor-node-6261f2f6-1.png](https://cloudius-jenkins-test.s3.amazonaws.com/6261f2f6-1050-47ad-93c5-02015b6b0721/20210225_172323/grafana-screenshot-overview-20210225_172323-longevity-cdc-100gb-4h-4-4-monitor-node-6261f2f6-1.png)
db-cluster - [https://cloudius-jenkins-test.s3.amazonaws.com/6261f2f6-1050-47ad-93c5-02015b6b0721/20210225_173156/db-cluster-6261f2f6.zip](https://cloudius-jenkins-test.s3.amazonaws.com/6261f2f6-1050-47ad-93c5-02015b6b0721/20210225_173156/db-cluster-6261f2f6.zip)
loader-set - [https://cloudius-jenkins-test.s3.amazonaws.com/6261f2f6-1050-47ad-93c5-02015b6b0721/20210225_173156/loader-set-6261f2f6.zip](https://cloudius-jenkins-test.s3.amazonaws.com/6261f2f6-1050-47ad-93c5-02015b6b0721/20210225_173156/loader-set-6261f2f6.zip)
monitor-set - [https://cloudius-jenkins-test.s3.amazonaws.com/6261f2f6-1050-47ad-93c5-02015b6b0721/20210225_173156/monitor-set-6261f2f6.zip](https://cloudius-jenkins-test.s3.amazonaws.com/6261f2f6-1050-47ad-93c5-02015b6b0721/20210225_173156/monitor-set-6261f2f6.zip)
sct-runner - [https://cloudius-jenkins-test.s3.amazonaws.com/6261f2f6-1050-47ad-93c5-02015b6b0721/20210225_173156/sct-runner-6261f2f6.zip](https://cloudius-jenkins-test.s3.amazonaws.com/6261f2f6-1050-47ad-93c5-02015b6b0721/20210225_173156/sct-runner-6261f2f6.zip)


[Jenkins job URL](https://jenkins.scylladb.com/job/scylla-4.4/job/longevity/job/longevity-cdc-4h-test/19/)

**Comment by tgrabiec** [comment]
This will be fixed after we transition schema changes to raft, because then we can pull the schema, and get apply all prior schema changes, before deserializing schema mutations.

**Comment by tgrabiec** [comment]
We could fix this before moving to raft if we invert the order in get_schema_for_write(): first sync, then call get_schema_definition()

**Comment by denesb** [comment]
Untouched for almost 4 years, closing as stale.
